modules/plugins/formatter/conform-nvim/conform-nvim.nix

```
@@ -30,27 +27,68 @@ in {
         description = "Default values when calling `conform.format()`";
       };
 
-      format_on_save = mkOption {
-        type = nullOr attrs;
-        default = {
+      format_on_save = let
+        defaultFormatOnSaveOpts = {
           lsp_format = "fallback";
           timeout_ms = 500;
         };
-        description = ''
-          Table that will be passed to `conform.format()`. If this
-          is set, Conform will run the formatter on save.
-        '';
-      };
+      in
+        mkOption {
+          type = nullOr (either attrs luaInline);
+          default =
+            mkLuaInline
+            # lua
+            ''
+              function()
+                if not vim.g.formatsave or vim.b.disableFormatSave then
+                  return
+                else
+                  return ${toLuaObject defaultFormatOnSaveOpts}
+                end
+              end
+            '';
+          description = ''
+            Table or function(lualinline) that will be passed to `conform.format()`. If this
+            is set, Conform will run the formatter on save.
 
-      format_after_save = mkOption {
-        type = nullOr attrs;
-        default = {lsp_format = "fallback";};
-        description = ''
-          Table that will be passed to `conform.format()`. If this
-          is set, Conform will run the formatter asynchronously after
-          save.
-        '';
-      };
+            Note:
+              - When config.vim.lsp.formatOnSave is set to true, internally
+                vim.g.formatsave is set to true.
+              - vim.b.disableFormatSave initally equals !config.vim.lsp.formatOnSave.
+              - vim.b.disableFormatSave is toggled using the
+                mapping from config.vim.lsp.mappings.toggleFormatOnSave.
```

```suggestion
            :::{.note}
              - When config.vim.lsp.formatOnSave is set to true, internally
                vim.g.formatsave is set to true.
              - vim.b.disableFormatSave initally equals !config.vim.lsp.formatOnSave.
              - vim.b.disableFormatSave is toggled using the
                mapping from config.vim.lsp.mappings.toggleFormatOnSave.
					  :::
```

Nixpkgs commonmark can handle note styles. Also put vim globals etc inside backticks.

---

>Nixpkgs commonmark can handle note styles. Also put vim globals etc inside backticks.

I just removed the section as @horriblename suggested removing the section as these are just internal APIs